### DISCUSSION TITLE
How do I add an offset to an events: state?  i.e. "1 hour before sunset"

### DISCUSSION THREAD

**Discussion by YAMLcase** [zachowj/node-red-contrib-home-assistant-websocket#932] [Q&A]
I'm using `sun.sun` as the entity, but would like it to trigger 1 hour before `below_horizon` is true.  How's the best way to accomplish this?

**Comment by zachowj** [comment]
I would recommend using the [node-red-contrib-chronos](https://flows.nodered.org/node/node-red-contrib-chronos) package for sun events.

If you still want to use the sun entity from HA you can use the time node using one of the `next_` attributes with the offset field.
